drivers/note/note_driver.c

```diff
@@ -1837,7 +1837,7 @@ void sched_note_filter_mode(FAR struct note_filter_named_mode_s *oldm,
   irqstate_t irq_mask;
   FAR struct note_driver_s **driver;
 
-  irq_mask = spin_lock_irqsave_wo_note(&g_note_lock);
+  irq_mask = spin_lock_irqsave_notrace(&g_note_lock);
```

Comment:
but nuttx trace framework name note not trace.

Reply:
But higher-level interfaces are trace models:
https://github.com/apache/nuttx/blob/master/include/nuttx/trace.h
could we rename trace framework from note to trace?

Reply:
either way is fine, but we need be consistent.

Reply:
I think this may require 2 stages to finish this work:
1. refine current code base, rename caller interface from note to trace.
2. rename note driver module to trace

Reply:
yes, I am fine with the proposal. please update the document too.

Reply:
@xiaoxiang781216 Please merge this PR if you do not have any comments

Reply:
`wo_` was added by Xiaomi. It is not in the history of nuttx. Some developers mistakenly believe that `wo` is the abbreviation of `write only`. The abbreviation wo(without) is not an English convention. I think we don't need to make any document changes.

Reply:
sorry, it isn't added by us, we just follow the original code base. The first patch come from:
```
commit ab3fa890239399de51e5482e595a751fee249b1c
Author: Masayuki Ishikawa <Masayuki.Ishikawa@jp.sony.com>
Date:   Wed Jan 17 13:08:03 2018 +0900

    SMP: Introduce spin_lock_wo_note() and spin_unlock_wo_note()
    
    These APIs are used in sched_note.c to protect instumentation data.
    The deffrence between these APIs to exsiting spin_lock() and spin_unlock()
    is that they do not perform insturumentation to avoid recursive call
    when SCHED_INSTRUMENTATION_SPINLOCKS=y.
```

Reply:
But why would you allow `trace.h` to be merged into nuttx? And the kernel code is also using similar APIs?

https://github.com/apache/nuttx/blob/master/include/nuttx/trace.h
https://github.com/apache/nuttx/blob/master/sched/task/task_init.c#L97
![image](https://github.com/user-attachments/assets/3457a742-1b67-4c3d-85a2-c3ea16c1a63d)

should we rename all trace to note?

Reply:
Yes, @Gary-Hobson please fix the inconsistent.